### ISSUE TITLE
Disabling the "Script Editor" Editor Feature makes it impossible to Detach Scripts

### ISSUE THREAD

**Issue by thygrrr** [godotengine/godot#84887]
### Godot version

4.2.dev

### System information

Windows 11

### Issue description

The "Detach the script from the selected node" icon is removed when disabling the "Script Editor" feature.

Existing and new scripts cannot be detached.

![image](https://github.com/godotengine/godot/assets/8904620/284ad55d-00b7-481e-8688-e5b5b229e58a)

The same is true for the "Detach Script" context menu item.

![image](https://github.com/godotengine/godot/assets/8904620/26b934d4-94fb-4287-afe9-63e5c31129bc)

However, since it's still possible to Attach a script via drag & drop, disabling this editor feature is detrimental to a development workflow and provides a dead end.

![image](https://github.com/godotengine/godot/assets/8904620/65023ba7-722d-42dc-a1c0-276a6a532df7)

New scripts can still be created in the FileSystem tab:
![image](https://github.com/godotengine/godot/assets/8904620/261aed0d-2c57-4550-90b5-72a19202199d)

But they cannot be created in the hierarchy.

Suggested fix: Keep the Detach Script functionality available under all circumstances, making it possible to have a clean workflow without the script editor always stealing focus. It would also be good to always be able to Create Scripts via context menu.

Dream fix for the future: In addition to suggested fix, clicking the script icons (or double-clicking script resources) would actually open them in the chosen external editor. (probably out of scope for the fix)

### Steps to reproduce

1. Open Editor->Manage Editor Features
2. Disable the "Script Editor" feature
3. Observe: Scripts can no longer be detached from their nodes.

### Minimal reproduction project

No "project" repro possible, Editor Features are managed somewhere outside the projects.

(actually would suggest to save that in project.godot - so the correct editor config can be loaded for a given project :))

**Comment by AThousandShips** [comment]
I think this might be a matter for discussion, one reason to disable scripting is for teaching, to restrict students in how they interact, preventing them from accidentally detaching a script makes a lot of sense to me

**Comment by thygrrr** [comment]
> I think this might be a matter for discussion, one reason to disable scripting is for teaching, to restrict students in how they interact, preventing them from accidentally detaching a script makes a lot of sense to me

Yes, I wanted to suggest a "pure fix" which would also disable attaching scripts. However, 99% of use cases, even during teaching, would at the very least involve attaching and detaching in some fashion. So it would miss the main reason people want to hide the script editor - to have a pure scene workflow.

However, there could be a "Scripting" editor feature that encompasses all of the above. Disabling "Script Editor" should disable the editing of scripts in said internal editor, I would say.
